x/interchainstaking/keeper/withdrawal_record.go

```diff
@@ -14,18 +16,37 @@ import (
 	"github.com/quicksilver-zone/quicksilver/x/interchainstaking/types"
 )
 
-func (k *Keeper) GetNextWithdrawalRecordSequence(ctx sdk.Context) (sequence uint64) {
-	store := prefix.NewStore(ctx.KVStore(k.storeKey), nil)
+// InitWithdrawalRecordSequence initializes the sequence.
+func (k *Keeper) InitWithdrawalRecordSequence(ctx sdk.Context) {
+	store := ctx.KVStore(k.storeKey)
+
+	bz := k.cdc.MustMarshal(&gogotypes.UInt64Value{Value: 512})
+	store.Set(types.KeyPrefixRequeuedWithdrawalRecordSeq, bz)
+}
+
+// GetNextWithdrawalRecordSequence returns and increments the global withdrawal record seqeuence.
+func (k *Keeper) GetNextWithdrawalRecordSequence(ctx sdk.Context) uint64 {
+	var sequence uint64
+	store := ctx.KVStore(k.storeKey)
+
 	bz := store.Get(types.KeyPrefixRequeuedWithdrawalRecordSeq)
 	if bz == nil {
-		bz := make([]byte, 8)
-		binary.BigEndian.PutUint64(bz, uint64(2))
-		store.Set(types.KeyPrefixRequeuedWithdrawalRecordSeq, bz)
-		return 1
+		// initialize the account numbers
+		sequence = 0
+	} else {
+		val := gogotypes.UInt64Value{}
+
+		err := k.cdc.Unmarshal(bz, &val)
+		if err != nil {
+			panic(err)
+		}
+
+		sequence = val.GetValue()
 	}
-	sequence = binary.BigEndian.Uint64(bz)
-	binary.BigEndian.PutUint64(bz, sequence+1)
+
+	bz = k.cdc.MustMarshal(&gogotypes.UInt64Value{Value: sequence + 1})
 	store.Set(types.KeyPrefixRequeuedWithdrawalRecordSeq, bz)
+
```

Comment:
I misread at first.  This is coming from OUR store, so I think that I'd prefer this function to panic.  If our store has corrupted data of any kind, quicksilver should go down.

Reply:
@coderabbitai please note that this style is what we call in cosmos "safety over liveness" -- we will often choose to panic so that the entire blockchain network goes down in the event that there is a critical issue. 

If it were consuming data from other chains, then I would not want it to panic.  Hope this makes sense.